src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -63,16 +58,19 @@ export const TradeConfirm = ({ history }: RouterProps) => {
     selectFirstAccountSpecifierByChainId(state, buyAssetChainId),
   )
 
-  const parsedTxId = useMemo(
-    () => serializeTxIndex(buyAssetAccountSpecifier, txid, trade.receiveAddress),
-    [buyAssetAccountSpecifier, trade.receiveAddress, txid],
+  const parsedBuyTxId = useMemo(
+    () => serializeTxIndex(buyAssetAccountSpecifier, buyTxid, trade.receiveAddress),
+    [buyAssetAccountSpecifier, trade.receiveAddress, buyTxid],
   )
-  const status = useAppSelector(state => selectTxStatusById(state, parsedTxId))
+
+  const status =
+    useAppSelector(state => selectTxStatusById(state, parsedBuyTxId)) ?? TxStatus.Pending
 
   const { showErrorToast } = useErrorHandler()
 
   const onSubmit = async () => {
     try {
+      setSellTxid('pending')
```

Comment:
repurposing this state variable to a loading status is a bug waiting to happen.

i can see the consumption below where you check for this string - can we please use another appropriate named state variable for this purpose?

Reply:
cleaned it up. please take another look